cmake/onnxruntime_common.cmake

```diff
@@ -56,19 +56,10 @@ if (onnxruntime_USE_TELEMETRY)
   set_target_properties(onnxruntime_common PROPERTIES COMPILE_FLAGS "/FI${ONNXRUNTIME_INCLUDE_DIR}/core/platform/windows/TraceLoggingConfigPrivate.h")
 endif()
 
-if (onnxruntime_USE_MIMALLOC)
-    if(onnxruntime_USE_CUDA OR onnxruntime_USE_OPENVINO) 
```

Comment:
What is the reason for getting rid of this guard in the ORT cmake file? If someone isn't using MIMALLOC why do we want to potentially pollute the build config by pulling in external/mimalloc.cmake?

Reply:
The original use of the --use_mimalloc/associated macros was to guard overriding the default base memory allocator used by python etc. this change is different in that we're not overriding the underlying memory allocator of python. 

It sounds like we might need to introduce a different switch as a guard for this case, something like --use_stl_allocator_mimalloc (or something less verbose). Would that be an acceptable alternative?

Reply:
Possibly. I think there should be a global on/off for whether any mimalloc stuff is included. If I'm not using mimalloc I shouldn't see any build files or code files being included. 

Inside of that global 'mimalloc enabled' guard I think you're free to add whatever other params you want.